src/Microsoft.Extensions.Options/OptionsService.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System;
+
+namespace Microsoft.Extensions.Options
+{
+    /// <summary>
+    /// Implementation of IOptionsFactory.
+    /// </summary>
+    /// <typeparam name="TOptions">The type of options being requested.</typeparam>
+    public class OptionsService<TOptions> : IOptionsService<TOptions> where TOptions : class, new()
```

Comment:
I'm fine with using factories, validators, configurators directly somewhere in the core or like API for providing custom things, but like API for using options - that's weird.

I agree that having different access patterns is also weird :)

But maybe we can figure out another options to allowing different accesses but with the same pattern? How about attributes?

Something like:

```
class ThatUsesOptions([OptionsNamed("name")] IOptions<TOptions> options)
class ThatUsesOptions([OptionsCached] IOptions<TOptions> options)
class ThatUsesOptions([OptionsSnapshot] IOptions<TOptions> options)
class ThatUsesOptions([OptionsNotCached] IOptions<TOptions> options)
```

What do you think?